File: LeetCodeChallenges/Arrays 101/DuplicateZeros.java
public class DuplicateZeros {
    public static void main(String[] args) {
        int[] arr = { 1, 0, 2, 3, 0, 4, 5, 0 };
        duplicateZeros(arr);
    }

    public static void duplicateZeros(int[] arr) {
        int temp = 0;
        for (int i = arr.length - 1; i >= 0; i--) {
            if(arr[i] == 0 && i < arr.length - 1){
                temp = i;
                rotateToTheRigh(arr, temp);
                arr[i] = 0;
            }

        }

    }

    public static void rotateToTheRigh(int[] arr, int n) {
        for (int j = arr.length - 1; j > n; j--) {
            arr[j] = arr[j - 1];
        }
    }
}

File: Arrays101/Inserting Item Into An Array/duplicatezeros.py
from typing import List
from collections import deque


def duplicateZeros(arr: List[int]) -> None:
    b = len(arr) - 2
    i = 0
    while i < b:
        if arr[i] == 0:
            arr.insert(i + 1, 0 )
            arr.pop(-1)
            i += 1
        i += 1



arr = [1,0,2,3,0,4,5,0]
duplicateZeros(arr)
print(arr)